### ISSUE TITLE
Regression in .NET Core SDK: This operation cannot be performed on a default instance of ImmutableArray

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi#8474]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

### Steps to reproduce

Checkout pulumi/examples, digitalocean-cs-loadbalanced-droplets, run `pulumi preview`.

Tweak csproj:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <Nullable>enable</Nullable>
    <AssemblyName>Digitalocean.LoadbalancedDroplets</AssemblyName>
    <RootNamespace>Digitalocean.LoadbalancedDroplets</RootNamespace>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Pulumi" Version="3.18.0" />
    <!-- <PackageReference Include="Pulumi" Version="3.17.1" /> -->
    <PackageReference Include="Pulumi.DigitalOcean" Version="4.8.0" />
  </ItemGroup>

</Project>
```

Expected: works the same as on 3.17.1
Actual: with Pulumi 3.18.0 fails as:

```
  pulumi:pulumi:Stack (digitalocean-cs-loadbalanced-droplets-dev):
    error: Running program '/Users/anton/examples/digitalocean-cs-loadbalanced-droplets/bin/Debug/netcoreapp3.1/Digitalocean.LoadbalancedDroplets.dll' failed with an unhandled exception:
    System.InvalidOperationException: >.  Consider initializing the array, or checking the ImmutableArray<T>.IsDefault property.
       at void System.Collections.Immutable.ImmutableArray<T>.ThrowInvalidOperationIfNotInitialized()
       at int System.Collections.Immutable.ImmutableArray<T>.System.Collections.ICollection.get_Count()
       at async Task<ImmutableArray<object>> Pulumi.Serialization.Serializer.SerializeListAsync(string ctx, IList list, bool keepResources, bool keepOutputValues)
       at async Task<object> Pulumi.Serialization.Serializer.SerializeAsync(string ctx, object prop, bool keepResources, bool keepOutputValues) x 3
       at async Task<RawSerializationResult> Pulumi.Deployment.SerializeFilteredPropertiesRawAsync(string label, IDictionary<string, object> args, Predicate<string> acceptKey, bool keepResources, bool keepOutputValues)
       at async Task<SerializationResult> Pulumi.Deployment.SerializeFilteredPropertiesAsync(string label, IDictionary<string, object> args, Predicate<string> acceptKey, bool keepResources, bool keepOutputValues)
       at async Task<PrepareResult> Pulumi.Deployment.PrepareResourceAsync(string label, Resource res, bool custom, bool remote, ResourceArgs args, ResourceOptions options)
       at async Task<(string urn, string id, Struct data, ImmutableDictionary<string, ImmutableHashSet<Resource>> dependencies)> Pulumi.Deployment.RegisterResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options)
       at async Task<(string urn, string id, Struct data, ImmutableDictionary<string, ImmutableHashSet<Resource>> dependencies)> Pulumi.Deployment.ReadOrRegisterResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options)
       at async Task Pulumi.Deployment.CompleteResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options, ImmutableDictionary<string, IOutputCompletionSource> completionSources)

```

**Comment by t0yv0** [comment]
This gets fixed when I revert https://github.com/pulumi/pulumi/pull/8316

**Comment by iwahbe** [comment]
This has been fixed in pulumi `3.18.1`, but is still filtering out to our providers. In the mean time, specifying `<PackageReference Include="Pulumi" Version="3.*" />` should force nuget to use the latest version of pulumi.
